File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l,r=0,len(nums)-1

        if(len(nums)==1):
            return nums[0]

        while(True):
            mid=(l+r)//2

            if(nums[0]<=nums[mid] and nums[mid+1]<=nums[-1]):
                return min(nums[0],nums[mid+1])
            
            elif nums[0]>nums[mid]:
                r=mid
            elif nums[mid+1]>nums[-1]:
                l=mid+1

File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> subset = new ArrayList<>();
        backtrack(nums,subset,res);
        return res;
    }

    private void backtrack(int[] nums, List<Integer> subset, List<List<Integer>> res)
    {
        if(subset.size()==nums.length)
        {
            res.add(new ArrayList<>(subset));
            return;
        }

        for(int num: nums)
        {
            if (subset.contains(num))
                continue;
            subset.add(num);
            backtrack(nums,subset,res);
            subset.remove(subset.size()-1);
        }
    }
}
